### PR TITLE
docs: remove ref to old docker compose config

### LINKED ISSUE #4828: Bad integration test instructions
When running:

```console
$ TEST_INTEGRATION=1 TEST_INFLUXDB_IOX_CATALOG_DSN="..." cargo test --workspace
```

The following tests:

```
    config::tests::test_reading_kafka
    config::tests::test_writing_kafka
    kafka::tests::test_batching
    kafka::tests::test_generic
    kafka::tests::test_offset_after_broken_message
    kafka::tests::test_setup_topic_race
```

Fail with:

```
---- kafka::tests::test_setup_topic_race stdout ----
thread 'kafka::tests::test_setup_topic_race' panicked at 'TEST_INTEGRATION is set which requires running integration tests, but KAFKA_CONNECT is not set. Please run Kafka, perhaps by using the command `docker-compose -f docker/ci-kafka-docker-compose.yml up kafka`, then set KAFKA_CONNECT to the host and port where Kafka is accessible. If running the `docker-compose` command and the Rust tests on the host, the value for `KAFKA_CONNECT` should be `localhost:9093`. If running the Rust tests in another container in the `docker-compose` network as on CI, `KAFKA_CONNECT` should be `kafka:9092`.', write_buffer/src/kafka/mod.rs:477:20
```

But the specified file does not exist:

```console
$ file  docker/ci-kafka-docker-compose.yml
docker/ci-kafka-docker-compose.yml: cannot open `docker/ci-kafka-docker-compose.yml' (No such file or directory)
```

### PR DESCRIPTION
**Pull request by domodwyer**
Closes #4828.

---

* docs: remove ref to old docker compose config (c337fe907)

      Remove references to the now-gone ci-kafka-docker-compose.yml command.

### CODE REVIEW

File: iox_data_generator/README.md [hunk 1]
```diff
@@ -51,14 +51,7 @@ cargo run --release -- run router --server-id 1
...
 
-You'll also need to run a Kafka instance. There's a Docker compose script in the influxdb_iox
-repo you can run with:
-
-```
-docker-compose -f docker/ci-kafka-docker-compose.yml up kafka
-```
-
-The Kafka instance will be accessible from `127.0.0.1:9093` if you run it with this script.
+You'll also need to run a Kafka instance separately.
```
Comment (crepererum) [inline iox_data_generator/README.md:54]: How?

Comment (domodwyer) [reply iox_data_generator/README.md:54]: I think we can expect users to know / be able to google how to set up kafka?

Seems redundant to give full instructions how to set up a kafka instance, not to mention it is prone to rotting...

**Comment by alamb** [comment]
Shall we clean up this PR and get it merged?

**Comment by crepererum** [comment]
I think we have a docker compose file again: https://github.com/influxdata/influxdb_iox/blob/main/integration-docker-compose.yml

**Pull request closed without merging**